plugins/com.google.cloud.tools.eclipse.sdk/src/com/google/cloud/tools/eclipse/sdk/CloudSdkProvider.java

```diff
@@ -28,17 +28,26 @@
 import java.nio.file.Path;
 
 /**
- * Utililty to find the Google Cloud SDK either at locations configured by the user or in standard
+ * Utility to find the Google Cloud SDK either at locations configured by the user or in standard
  * locations on the system.
  */
 public class CloudSdkProvider extends ContextFunction {
 
+  private final IPreferenceStore preferences;
+
+   public CloudSdkProvider(IPreferenceStore preferences) {
```

Comment:
I'd prefer to have an argumentless constructor that works as the `null` case instead of having constructor calls with `null`s and then probably adding a comment with the argument name.